core/processing.go

```diff
@@ -33,7 +76,63 @@ func ProduceBlackWhite(inputImg string) {
 		}
 	}
 
-	outputFile, err := os.Create("../dist/output.png")
+	outputFile, err := os.Create("./dist/gray_output.png")
+	if err != nil {
+		panic(err)
+	}
+	defer outputFile.Close()
+
+	err = png.Encode(outputFile, newImg)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func Resize(inputImg string, factor int) {
+	file, err := os.Open(inputImg)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	img, _, err := image.Decode(file)
+	if err != nil {
+		panic(err)
+	}
+
+	newImg := image.NewRGBA(image.Rect(0, 0, img.Bounds().Max.X*factor, img.Bounds().Max.Y*factor))
+
+	otherdraw.NearestNeighbor.Scale(newImg, newImg.Rect, img, img.Bounds(), otherdraw.Over, nil)
+
+	outputFile, err := os.Create("./dist/scaled_output.png")
+	if err != nil {
+		panic(err)
+	}
+	defer outputFile.Close()
+
+	err = png.Encode(outputFile, newImg)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ResizeAbsolute(inputImg string, length int, width int) {
+	file, err := os.Open(inputImg)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	img, _, err := image.Decode(file)
```

Comment:
Would be good to extract file interactions (open, encode, decode) into a different domain